coordinator/src/services/translation/request/withdraw/withdraw.ts

```diff
@@ -1,17 +1,38 @@
-import {fhir, hl7V3} from "@models"
+import {fhir, hl7V3, processingErrors} from "@models"
 import {getCodeableConceptCodingForSystem, getIdentifierValueForSystem, getMessageId} from "../../common"
 import {convertIsoDateTimeStringToHl7V3DateTime} from "../../common/dateTime"
 import {getMessageIdFromTaskFocusIdentifier, getPrescriptionShortFormIdFromTaskGroupIdentifier} from "../task"
 import Hapi from "@hapi/hapi"
 import {getSdsRoleProfileId, getSdsUserUniqueId} from "../../../../utils/headers"
+import {getContainedPractitionerRoleViaReference} from "../../common/getResourcesOfType"
+import {isReference} from "../../../../../src/utils/type-guards"
+import {createAuthorForWithdraw} from "../agent-unattended"
 
-export function convertTaskToEtpWithdraw(task: fhir.Task, headers: Hapi.Util.Dictionary<string>): hl7V3.EtpWithdraw {
+export function convertTaskToEtpWithdraw(task: fhir.Task): hl7V3.EtpWithdraw {
   const id = getMessageId(task.identifier, "Task.identifier")
   const effectiveTime = convertIsoDateTimeStringToHl7V3DateTime(task.authoredOn, "Task.authoredOn")
   const etpWithdraw = new hl7V3.EtpWithdraw(new hl7V3.GlobalIdentifier(id), effectiveTime)
 
+  const practitionerRoleRef = task.requester
+  if(!isReference(practitionerRoleRef)) {
+    throw new processingErrors.InvalidValueError(
+      "task.requester should be a reference to Practitioner Role",
```

Comment:
could be a bit more specific with this and say `... reference to contained.practitionerRole` or something similar